Fix intermittent test failure in `TestRepackFileUpload.test_repacking_xpi_files`

`TestRepackFileUpload.test_repacking_xpi_files` is failing intermittently with:

```
    assert tmp_dir_before_repacking == sorted(os.listdir(tempfile.gettempdir()))
E   AssertionError: assert ['.bash_env-636a60479ec4987fb8cecebf-0-build', 'circle-agent-runner.pid', 'circleci-1031628640', 'circleci-ts.sock', 'pytest-of-circleci', 'python-build.20221103012145.98.log', 'stdbool.hkqzzv063.c', 'tmpnlr718cn.zip'] == ['.bash_env-636a60479ec4987fb8cecebf-0-build', 'circle-agent-runner.pid', 'circleci-1031628640', 'circleci-ts.sock', 'pytest-of-circleci', 'python-build.20221103012145.98.log', 'stdbool.hkqzzv063.c', 'tmpkcq1qdnk.zip', 'tmpnlr718cn.zip']
E     At index 7 diff: 'tmpnlr718cn.zip' != 'tmpkcq1qdnk.zip'
E     Right contains one more item: 'tmpnlr718cn.zip'
E     Full diff:
E       [
E        '.bash_env-636a60479ec4987fb8cecebf-0-build',
E        'circle-agent-runner.pid',
E        'circleci-1031628640',
E        'circleci-ts.sock',
E        'pytest-of-circleci',
E        'python-build.20221103012145.98.log',
E        'stdbool.hkqzzv063.c',
E     -  'tmpkcq1qdnk.zip',
E        'tmpnlr718cn.zip',
E       ]
```

A recent example: https://app.circleci.com/pipelines/github/mozilla/addons-server/23956/workflows/c2280a68-fdf6-4688-bc2d-8523d95bfbf7/jobs/140152

This is because it expect the contents of `tempfile.gettempdir()` to only be affected by the test... but we're running tests in parallel with `pytest-xdist`, and `tempfile.tempdir` remains the same even during parallel executions of tests (it's simply `/tmp`).

It's intermittent because depending on test ordering and speed of execution, the test might run without anything else being written to to that temporary directory, or because the other test writing to it might have had time to delete anything it wrote.

The solution is to ensure each test in this class (or globally?) is using its own `tempfile.tempdir`.

--- a/src/olympia/files/tests/test_tasks.py
+++ b/src/olympia/files/tests/test_tasks.py
@@ -3,6 +3,7 @@ import os
 import tempfile
 import zipfile
 
+import pytest
 from unittest import mock
 from waffle.testutils import override_switch
 
@@ -16,6 +17,13 @@ from olympia.files.tests.test_utils import AppVersionsMixin
 
 
 class TestRepackFileUpload(AppVersionsMixin, UploadMixin, TestCase):
+    @pytest.fixture(autouse=True)
+    def set_tempfile_tempdir_for_class(self, tmp_path):
+        original = tempfile.tempdir
+        tempfile.tempdir = str(tmp_path)
+        yield
+        tempfile.tempdir = original
+
     @mock.patch('olympia.amo.utils.SafeStorage.move_stored_file')
     @mock.patch('olympia.files.tasks.get_sha256')
     @mock.patch('olympia.files.tasks.shutil')

--- a/src/olympia/files/tests/test_tasks.py
+++ b/src/olympia/files/tests/test_tasks.py
@@ -19,7 +19,18 @@ from olympia.files.tests.test_utils import AppVersionsMixin
 class TestRepackFileUpload(AppVersionsMixin, UploadMixin, TestCase):
     @pytest.fixture(autouse=True)
     def set_tempfile_tempdir_for_class(self, tmp_path):
+        """Fixture setting tempfile.tempdir for each test in this class,
+        allowing them to run in parallel each with their own tempdir.
+
+        It reverts back to the original tempdir after each test.
+
+        repack_fileupload() depends on tempfile methods which are using
+        tempfile.tempdir under the hood, and the tests expect to be able to
+        compare the contents of that directory to check no files were left
+        behind.
+        """
         original = tempfile.tempdir
+        # tmp_path is automatically provided by pytest (it's a fixture).
         tempfile.tempdir = str(tmp_path)
         yield
         tempfile.tempdir = original

